### PR TITLE
Hyperlink fixes

### PR DESCRIPTION
**Pull request by philrz**
## tl;dr

Make some link checker failures go away.

## Details

I made a mistake in one of my aspirational "it'll start working once it merges" hyperlinks in #5506, so I tried fixing that here. That revealed that the markdown link checker we've been using also seems to have some kind of bug with hyperlinks containing [percent encoding](https://en.wikipedia.org/wiki/Percent-encoding). I just added a [comment](https://github.com/gaurav-nelson/github-action-markdown-link-check/issues/102#issuecomment-2504463659) to an existing issue with a simple repro example, but for now I've added an exclusion in our link checker config. This also led me to notice that the guy who maintained that markdown link checker has a new tool [linkspector](https://github.com/UmbrellaDocs/linkspector) that seems to be more actively maintained, so I'll separately look into whether we can move over to that one and maybe not have to do exclusions.

While I'm at it, I'm also adjusting the hyperlink for an Apache Hive reference. Their design docs happen to be on a public Confluence site and it must not be built for scale because it seems like 50% of the time our link checker experiences some kind of timeout trying to crawl it. Rather than drop the hyperlink entirely or add an exclusion to the link checker I'm opting to just point to something one link further back in their "real" web site.

### CODE REVIEW

**Comment by philrz** [comment]
It looks like moving to a different link checker may be a better fix here. Closing this one without merging.
